### DISCUSSION TITLE
how to make custom loading?

### DISCUSSION THREAD

**Discussion by BigNeutronStar** [flet-dev/flet#3453] [Q&A] [answered]
### Question

how to make custom loading (i have processRing now)? I want to use gif as loading in my flet project. What should I do?

### Code sample

```python
class Loading(Page):
    def __init__(self, page: ft.Page):
        super().__init__(page)
        self.page = page
        self.controls = [
            ft.Container(
                content=ft.Row(
                    [
                        ft.Column(
                            [
                                ft.ProgressRing(),
                                ft.Text("loading"),
                            ],
                            alignment=ft.MainAxisAlignment.CENTER,
                            horizontal_alignment=ft.CrossAxisAlignment.CENTER,
                            expand=True,
                        )
                    ],
                    alignment=ft.MainAxisAlignment.CENTER,
                    vertical_alignment=ft.CrossAxisAlignment.CENTER,
                    expand=True,
                ),
                alignment=ft.alignment.center,
                expand=True,
            )
        ]
```

### Error message

```bash
i don't have an error
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment, marked as answer]
Use the `Image` Control.
I think it supports gif files.

**Reply by BigNeutronStar** [reply]
image don't support gifs :(

**Reply by ndonkoHenri** [reply]
It normally should. I can't try out rn, but remember I tried that someday and it worked.
Can you share a link to the gif you are trying to view?

**Reply by BigNeutronStar** [reply]
i can't share a link, but i can upload it in repository in my github, i'll name this repository GIF

**Reply by BigNeutronStar** [reply]
here's a link on gif: https://github.com/BigNeutronStar/GIF

**Reply by ndonkoHenri** [reply]
The following code works:

```py
import flet as ft


def main(page: ft.Page):
    page.add(
        ft.Image(
            src="https://github.com/BigNeutronStar/GIF/blob/main/custom_loading.gif?raw=true"
        )
    )


ft.app(target=main)
```

![xx-ezgif com-video-to-gif-converter (3)](https://github.com/flet-dev/flet/assets/98978078/4db4335a-5caa-4ad6-853f-71fcf5fa8b04)

**Reply by BigNeutronStar** [reply]
bro i'll test it on my pc but thank you so much!

**Reply by BigNeutronStar** [reply]
> The following code works:
> 
> ```python
> import flet as ft
> 
> 
> def main(page: ft.Page):
>     page.add(
>         ft.Image(
>             src="https://github.com/BigNeutronStar/GIF/blob/main/custom_loading.gif?raw=true"
>         )
>     )
> 
> 
> ft.app(target=main)
> ```
> 
> ![xx-ezgif com-video-to-gif-converter (3)](https://private-user-images.githubusercontent.com/98978078/338949905-4db4335a-5caa-4ad6-853f-71fcf5fa8b04.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxOTI5OTIsIm5iZiI6MTcxODE5MjY5MiwicGF0aCI6Ii85ODk3ODA3OC8zMzg5NDk5MDUtNGRiNDMzNWEtNWNhYS00YWQ2LTg1M2YtNzFmY2Y1ZmE4YjA0LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDExNDQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0ZDkxM2M4YzlhMzk1ZmQ5Yzg0YTgzYjMxNGMzNmM1YjY0NWFjYmYzMGU2ZmFjZTEwNjY1NjA3N2VjNDgzZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.euE8o3yw8hgbJN4fCrFsuRg_PjSnaPcp3OlmEvqXtVY) [ ![xx-ezgif com-video-to-gif-converter (3)](https://private-user-images.githubusercontent.com/98978078/338949905-4db4335a-5caa-4ad6-853f-71fcf5fa8b04.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxOTI5OTIsIm5iZiI6MTcxODE5MjY5MiwicGF0aCI6Ii85ODk3ODA3OC8zMzg5NDk5MDUtNGRiNDMzNWEtNWNhYS00YWQ2LTg1M2YtNzFmY2Y1ZmE4YjA0LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDExNDQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0ZDkxM2M4YzlhMzk1ZmQ5Yzg0YTgzYjMxNGMzNmM1YjY0NWFjYmYzMGU2ZmFjZTEwNjY1NjA3N2VjNDgzZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.euE8o3yw8hgbJN4fCrFsuRg_PjSnaPcp3OlmEvqXtVY) ](https://private-user-images.githubusercontent.com/98978078/338949905-4db4335a-5caa-4ad6-853f-71fcf5fa8b04.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxOTI5OTIsIm5iZiI6MTcxODE5MjY5MiwicGF0aCI6Ii85ODk3ODA3OC8zMzg5NDk5MDUtNGRiNDMzNWEtNWNhYS00YWQ2LTg1M2YtNzFmY2Y1ZmE4YjA0LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDExNDQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0ZDkxM2M4YzlhMzk1ZmQ5Yzg0YTgzYjMxNGMzNmM1YjY0NWFjYmYzMGU2ZmFjZTEwNjY1NjA3N2VjNDgzZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.euE8o3yw8hgbJN4fCrFsuRg_PjSnaPcp3OlmEvqXtVY) [ ](https://private-user-images.githubusercontent.com/98978078/338949905-4db4335a-5caa-4ad6-853f-71fcf5fa8b04.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxOTI5OTIsIm5iZiI6MTcxODE5MjY5MiwicGF0aCI6Ii85ODk3ODA3OC8zMzg5NDk5MDUtNGRiNDMzNWEtNWNhYS00YWQ2LTg1M2YtNzFmY2Y1ZmE4YjA0LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDExNDQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0ZDkxM2M4YzlhMzk1ZmQ5Yzg0YTgzYjMxNGMzNmM1YjY0NWFjYmYzMGU2ZmFjZTEwNjY1NjA3N2VjNDgzZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.euE8o3yw8hgbJN4fCrFsuRg_PjSnaPcp3OlmEvqXtVY)

listen, I tried to add it to the project, the gif appeared, but errors also appeared related to other modules of the project (and it’s not clear why they appeared, because before the changes everything worked fine and the errors were not related to the gif)

there is class for loading:
class Loading(Page):
    def __init__(self, page: ft.Page):
        super().__init__(page)
        self.page = page
        self.controls = [
            ft.Container(
                content=ft.Row(
                    [
                        ft.Column(
                            [
                                ft.Image(
                                    src="https://github.com/BigNeutronStar/GIF/blob/main/custom_loading.gif?raw=true",
                                    width=100,
                                    height=100,
                                    fit=ft.ImageFit.CONTAIN,
                                ),
                                ft.Text("loading"),
                            ],
                            alignment=ft.MainAxisAlignment.CENTER,
                            horizontal_alignment=ft.CrossAxisAlignment.CENTER,
                            expand=True,
                        )
                    ],
                    alignment=ft.MainAxisAlignment.CENTER,
                    vertical_alignment=ft.CrossAxisAlignment.CENTER,
                    expand=True,
                ),
                alignment=ft.alignment.center,
                expand=True,
            )
        ]
        
and i have this error:

<img width="1425" alt="image" src="https://github.com/flet-dev/flet/assets/109436275/91191f1b-61c7-4d84-a47b-4aa00cc98cbc">

**Reply by BigNeutronStar** [reply]
stop. It is my error in databases, i'll try to fix it
